### DISCUSSION TITLE
Inconsistence in tutorials for scVelo?

### DISCUSSION THREAD

**Discussion by denvercal1234GitHub** [theislab/scvelo#1064] [Q&A] [answered]
Hi there, 

Thanks for the tool. 

In `scVelo` tutorial (https://scvelo.readthedocs.io/en/stable/DynamicalModeling/), we do not have `sc.tl.pca` and `sc.pp.neighbors` between filtering and moments. 

```
scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
scv.pp.moments(adata, n_pcs=30, n_neighbors=30)
```

However, in the `cellrank` tutorial during the `scVelo section` (https://cellrank.readthedocs.io/en/stable/cellrank_basics.html), it specified to run as follow. Would you mind letting me know do we need to run `sc.tl.pca` and `sc.pp.neighbors`?

```
scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
sc.tl.pca(adata)
sc.pp.neighbors(adata, n_pcs=30, n_neighbors=30)
scv.pp.moments(adata, n_pcs=None, n_neighbors=None)
```

**Comment by WeilerP** [comment, marked as answer]
The functions are called internally when calling `scv.pp.moments`.
